src/Symfony/Component/Notifier/Bridge/Twitter/TwitterOptions.php

```diff
@@ -0,0 +1,190 @@
+    }
+
+    public function getRecipientId(): ?string
+    {
+        return null;
+    }
+
+    /**
+     * @param string[] $options
+     *
+     * @return $this
+     */
+    public function pool(array $options, int $duration): static
+    {
+        $this->options['poll'] = [
+            'options' => $options,
+            'duration_minutes' => $duration,
+        ];
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
+    public function quote(string $id): static
+    {
+        $this->options['quote_tweet_id'] = $id;
+
+        return $this;
+    }
+
+    /**
+     * @param string[] $exclude
+     *
+     * @return $this
+     */
+    public function inReplyTo(string $id, array $exclude = []): static
+    {
+        $this->options['reply']['in_reply_to_tweet_id'] = $id;
+        $this->options['reply']['exclude_reply_user_ids'] = $exclude;
+
+        return $this;
+    }
+
+    /**
+     * @param string[] $extraOwners
+     *
+     * @return $this
+     */
+    public function attachImage(File $file, string $alt = '', array $extraOwners = []): static
+    {
+        $this->options['attach'][] = [
+            'file' => $file,
+            'alt' => $alt,
+            'subtitles' => null,
+            'category' => 'tweet_image',
+            'owners' => $extraOwners,
+        ];
+
+        return $this;
+    }
+
+    /**
+     * @param string[] $extraOwners
+     *
+     * @return $this
+     */
+    public function attachGif(File $file, string $alt = '', array $extraOwners = []): static
+    {
+        $this->options['attach'][] = [
+            'file' => $file,
+            'alt' => $alt,
+            'subtitles' => null,
+            'category' => 'tweet_gif',
+            'owners' => $extraOwners,
+        ];
+
+        return $this;
+    }
+
+    /**
+     * @param File     $subtitles   File should be named as "display_name.language_code.srt"
+     * @param string[] $extraOwners
+     *
+     * @return $this
+     */
+    public function attachVideo(File $file, string $alt = '', File $subtitles = null, bool $amplify = false, array $extraOwners = []): static
+    {
+        $this->options['attach'][] = [
+            'file' => $file,
+            'alt' => $alt,
+            'subtitles' => $subtitles,
+            'category' => $amplify ? 'amplify_video' : 'tweet_video',
+            'owners' => $extraOwners,
+        ];
+
+        return $this;
+    }
+
+    /**
+     * @param self::REPLY_* $settings
+     *
+     * @return $this
+     */
+    public function replySettings(string $settings): static
+    {
+        $this->options['reply_settings'] = $settings;
+
+        return $this;
+    }
+
+    /**
+     * @param string[] $ids
+     *
+     * @return $this
+     */
+    public function taggedUsers(string $ids): static
+    {
+        $this->options['media']['tagged_user_ids'] = $ids;
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
+    public function deepLink(string $url): static
```

Comment:
should we use `?string` to allow removing it ?

Same in many other setters ?

Reply:
I'd prefer not adding ways to unset. It's a minor use case that's not worth the complexity. One could create a new options instance to remove things if they want to.